src/Uno.UI.Runtime.Skia.MacOS/UnoNativeMac/UnoNativeMac/UNOWindow.m

```diff
@@ -165,11 +165,185 @@ void uno_window_set_min_size(NSWindow *window, double width, double height)
     window.minSize = CGSizeMake(width, height);
 }
 
+char* uno_window_get_title(NSWindow *window)
+{
+    return strdup(window.title.UTF8String);
+}
+
 void uno_window_set_title(NSWindow *window, const char* title)
 {
     window.title = [NSString stringWithUTF8String:title];
 }
 
+bool uno_window_is_full_screen(NSWindow *window)
+{
+    bool result = window != nil;
+    if (result) {
+        result = (window.styleMask & NSWindowStyleMaskFullScreen) == NSWindowStyleMaskFullScreen;
+    }
+#if DEBUG
+    NSLog(@"uno_window_is_full_screen %@ %s", window, result ? "true" : "false");
+#endif
+    return result;
+}
+
+bool uno_window_enter_full_screen(NSWindow *window)
+{
+    bool result = window != nil;
+    if (result && (window.styleMask & NSWindowStyleMaskFullScreen) != NSWindowStyleMaskFullScreen) {
+        [window toggleFullScreen:nil];
+        result = true;
+    }
+#if DEBUG
+    NSLog(@"uno_window_enter_full_screen %@ %s", window, result ? "true" : "false");
+#endif
+    return result;
+}
+
+void uno_window_exit_full_screen(NSWindow *window)
+{
+    if (window && (window.styleMask & NSWindowStyleMaskFullScreen) == NSWindowStyleMaskFullScreen) {
+        [window toggleFullScreen:nil];
+    }
+#if DEBUG
+    NSLog(@"uno_window_exit_full_screen %@", window);
+#endif
+}
+
+// on macOS double-clicking on the titlebar maximize the window (not the green icon)
+void uno_window_maximize(NSWindow *window)
+{
+#if DEBUG
+    NSLog(@"uno_window_maximize %@", window);
+#endif
+    [window performZoom:nil];
+}
+
+void uno_window_minimize(NSWindow *window, bool activateWindow)
+{
+#if DEBUG
+    NSLog(@"uno_window_minimize %@ %s", window, activateWindow ? "true" : "false");
+#endif
+    [window miniaturize:nil];
+    if (activateWindow) {
+        [window makeMainWindow];
+    }
+}
+
+void uno_window_restore(NSWindow *window, bool activateWindow)
+{
+#if DEBUG
+    NSLog(@"uno_window_restore %@ %s", window, activateWindow ? "true" : "false");
+#endif
+    switch(uno_window_get_overlapped_presenter_state(window)) {
+        case OverlappedPresenterStateMaximized:
+            [window zoom:nil];
+            break;
+        case OverlappedPresenterStateMinimized:
+            [window deminiaturize:nil];
+            break;
+        default:
```

Comment:
Should we NSLog in situations where we hit this default case?

Reply:
It was not useful to me 😄 

The `NSLog` are only present in `DEBUG` builds and the nuget only ship the release version of the binary. IOW the `NSLog` is only helpful if you build the native library yourself. Anything that needs to be logged is done from the managed assembly.